odoo/addons/base/models/ir_mail_server.py

```diff
@@ -181,36 +205,69 @@ def connect(self, host=None, port=None, user=None, password=None, encryption=Non
            :param user: optional username to authenticate with
            :param password: optional password to authenticate with
            :param string encryption: optional, ``'ssl'`` | ``'starttls'``
+           :param smtp_from: FROM SMTP envelop, used to find the best mail server
+           :param ssl_certificate: filename of the SSL certificate used for authentication
+               Used when no mail server is given and overwrite  the odoo-bin argument "smtp_ssl_certificate"
```

Comment:
Would be a good idea to check (and document) what kind of paths `load_cert_chain` allows here e.g. only absolute? Relative to cwd? With symbolic user component? With envvars?

And / or check for that when parsing the config. It extremely frustrating when things either don't work and don't provide any feedback (e.g. if `load_cert_chain` ignores "invalid" paths) or trigger errors which aren't relatable to what you are doing.

Reply:
Now filenames are only used with the odoo-bin argument.
Now you have a different error message if it can not found your file or if the certificate / private key is not valid,
and you know exactly what happen.

![screenshot_spect](https://user-images.githubusercontent.com/54316226/128353296-ea6987eb-3443-458a-9fd0-6c4d1de00ec2.png)

Reply:
(and it can only take absolute path)